lighthouse-treemap/app/src/main.js

```diff
@@ -9,6 +9,242 @@
+    const treemapDebugData = /** @type {LH.Audit.Details.DebugData} */ (
+      options.lhr.audits['script-treemap-data'].details);
+    if (!treemapDebugData || !treemapDebugData.treemapData) {
+      throw new Error('missing script-treemap-data');
+    }
+
+    /** @type {import('../../../lighthouse-core/audits/script-treemap-data').TreemapData} */
+    const scriptRootNodes = treemapDebugData.treemapData;
+
+    /** @type {WeakMap<LH.Treemap.Node, LH.Treemap.RootNodeContainer>} */
+    this.nodeToRootNodeMap = new WeakMap();
+
+    /** @type {{[group: string]: LH.Treemap.RootNodeContainer[]}} */
+    this.rootNodesByGroup = {
+      scripts: scriptRootNodes,
+    };
+
+    for (const rootNodes of Object.values(this.rootNodesByGroup)) {
+      for (const rootNode of rootNodes) {
+        TreemapUtil.walk(rootNode.node, node => this.nodeToRootNodeMap.set(node, rootNode));
+      }
+    }
+
+    /** @type {LH.Treemap.Node} */
+    this.currentRootNode; // eslint-disable-line no-unused-expressions
+    this.documentUrl = options.lhr.requestedUrl;
+    this.el = el;
+    this.getHueForKey = TreemapUtil.stableHasher(TreemapUtil.COLOR_HUES);
+
+    this.createHeader();
+    this.show();
+    this.initListeners();
+  }
+
+  createHeader() {
+    const urlEl = /** @type {HTMLAnchorElement} */ (TreemapUtil.find('.lh-header--url'));
+    urlEl.textContent = this.documentUrl;
+    urlEl.href = this.documentUrl;
+    TreemapUtil.find('.lh-header--size').textContent =
+      TreemapUtil.formatBytes(this.createRootNodeForGroup('scripts').resourceBytes);
+  }
+
+  initListeners() {
+    window.addEventListener('resize', () => {
+      this.resize();
+    });
+
+    const treemapEl = TreemapUtil.find('.lh-treemap');
+    treemapEl.addEventListener('click', (e) => {
+      if (!(e.target instanceof HTMLElement)) return;
+      const nodeEl = e.target.closest('.webtreemap-node');
+      if (!nodeEl) return;
+      this.updateColors();
+    });
+
+    treemapEl.addEventListener('mouseover', (e) => {
+      if (!(e.target instanceof HTMLElement)) return;
+      const nodeEl = e.target.closest('.webtreemap-node');
+      if (!nodeEl) return;
+      nodeEl.classList.add('webtreemap-node--hover');
+    });
+
+    treemapEl.addEventListener('mouseout', (e) => {
+      if (!(e.target instanceof HTMLElement)) return;
+      const nodeEl = e.target.closest('.webtreemap-node');
+      if (!nodeEl) return;
+      nodeEl.classList.remove('webtreemap-node--hover');
+    });
+  }
+
+  /**
+   * Combines all the root nodes for a given group into a new root node.
+   * @param {string} group
+   * @return {LH.Treemap.Node}
+   */
+  createRootNodeForGroup(group) {
+    const rootNodes = this.rootNodesByGroup[group];
+
+    const children = rootNodes.map(rootNode => {
+      // TODO: keep?
+      // Wrap with the name of the rootNode. Only for bundles.
+      if (group === 'scripts' && rootNode.node.children) {
+        return {
+          name: rootNode.name,
```

Comment:
calling out another naming thing that makes this hard. `LH.Treemap.RootNodeContainer` has a `name`  and `LH.Treemap.Node` has a name. they serve different purposes.

going along with my above rename, i can see former changing to `url` or `fileName`. (this'll require some treemap-script-data changes as well.)